src/javascript/app_2/Modules/Portfolio/Containers/portfolio.jsx

```diff
@@ -1,256 +1,89 @@
-import React                     from 'react';
-import PortfolioCard             from '../Components/portfolio_card.jsx';
-import { formatPortfolioData }   from '../Helpers/process_data';
-import DataTable                 from '../../../App/Components/Elements/data_table.jsx';
-import Tooltip                   from '../../../App/Components/Elements/tooltip.jsx';
-import { contract_type_display } from '../../../Constants/contract';
-import { WS }                    from '../../../Services';
-import { ProcessData }           from '../../../Services/Helpers';
-import { getAppId }              from '../../../../config';
-import ClientBase                from '../../../../_common/base/client_base';
-import { formatMoney }           from '../../../../_common/base/currency_base';
-import { localize }              from '../../../../_common/localize';
-import { getPropertyValue }      from '../../../../_common/utility';
-import Loading                   from '../../../../../templates/_common/components/loading.jsx';
-
-const app_id = getAppId();
-
-class Portfolio extends React.Component  {
+import classnames                     from 'classnames';
+import { PropTypes as MobxPropTypes } from 'mobx-react';
+import PropTypes                      from 'prop-types';
+import React                          from 'react';
+import CardList                       from '../Components/card_list.jsx';
+import EmptyPortfolioMessage          from '../Components/empty_portfolio_message.jsx';
+import { getTableColumnsTemplate }    from '../Constants/data_table_constants';
+import DataTable                      from '../../../App/Components/Elements/data_table.jsx';
+import { connect }                    from '../../../Stores/connect';
+import ClientBase                     from '../../../../_common/base/client_base';
+import Loading                        from '../../../../../templates/_common/components/loading.jsx';
+
+class Portfolio extends React.Component {
     state = {
-        currency   : ClientBase.get('currency').toLowerCase(),
-        data_source: [],
-        error      : null,
-        footer     : {
-            reference : 'Total',
-            payout    : '',
-            purchase  : '',
-            indicative: '',
-        },
-        is_loading: true,
-        oauth_apps: null,
+        // TODO: get currency from store, once it has been added
+        currency: ClientBase.get('currency').toLowerCase(),
```

Comment:
@ruskakimov 
IMO, it's better to define the `currency` property in `ClientStore` and pass to the component by `connect` instead of adding local state.

Reply:
that's what TODO stands for 😅

Reply:
@ruskakimov 
it has been added to the `ClientStore` by @kellybinary. You can fetch the last version of `new-app`